contracts/interchain-token-service/tests/interchain_transfer.rs

```diff
@@ -107,20 +112,17 @@ fn interchain_transfer_receive() {
 
     gateway_client.approve_messages(&messages, &proof);
 
+    let expected_event = InterchainTransferReceivedEvent {
+        topic1: Symbol::new(&env, "interchain_transfer_received"),
+        topic2: String::from_str(&env, HUB_CHAIN),
+        topic3: token_id,
+        topic4: sender,
+        topic5: recipient,
+        topic6: amount,
+        data1: data,
+    };
+
     client.execute(&source_chain, &message_id, &source_address, &payload);
 
-    assert_emitted_event(
-        &env,
-        -2,
-        &client.address,
-        (
-            Symbol::new(&env, "interchain_transfer_received"),
-            String::from_str(&env, HUB_CHAIN),
-            token_id,
-            sender,
-            recipient,
-            amount,
-        ),
-        (data,),
-    );
+    assert!(expected_event.matches(&env, &env.events().all().last().unwrap()));
```

Comment:
see existing usage with `fmt_last_emitted_event` for how we should test the event